WWDC/AppCoordinator.swift

```diff
@@ -248,32 +251,49 @@ final class AppCoordinator: Logging {
         DownloadManager.shared.syncWithFileSystem()
     }
 
+    var hasPerformedInitialListUpdate = false
     private func doUpdateLists() {
 
         // Initial app launch waits for all of these things to be loaded before dismissing the primary loading spinner
         // It may, however, delay the presentation of content on tabs that already have everything they need
 
-        let startupDependencies = Observable.combineLatest(storage.tracksObservable,
-                                                          storage.eventsObservable,
-                                                          storage.focusesObservable,
-                                                          storage.scheduleObservable)
+        let startupDependencies = Publishers.CombineLatest4(
+            storage.tracksObservable,
+            storage.eventsObservable,
+            storage.focusesObservable,
+            storage.scheduleObservable
+        )
 
         startupDependencies
+            .replaceErrorWithEmpty()
             .filter {
                 !$0.0.isEmpty && !$0.1.isEmpty && !$0.2.isEmpty
             }
-            .take(1)
-            .subscribe(onNext: { [weak self] tracks, _, _, sections in
-                guard let self = self else { return }
+            .prefix(1)
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] tracks, _, _, sections in
+                guard let self else { return }
 
                 self.tabController.hideLoading()
-                self.searchCoordinator.configureFilters()
+                if !hasPerformedInitialListUpdate {
+                    // Filters only need configured once, the other stuff in
+                    // here might only need to happen once as well
+                    self.searchCoordinator.configureFilters()
```

Comment:
This was causing filters to be reset after the content sync, I put a gate around it for now, but the long term solution is different than this.

Reply:
OMG thanks, this has been driving me nuts!

Reply:
The original reason it's always called is because it uses values from Storage. But the 1st setup and the update need to be separated so that they can take any user modifications into account after the first load. By blocking this, we're potentially missing an update to filtering capabilities, but that's a small thing for now!